src/Handler/TagHandler.php

```diff
@@ -50,6 +55,29 @@ public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cach
     }
 
     /**
+     * Add tags to be sent.
+     *
+     * This must be called before any response is sent to the client.
+     *
+     * @param array $tags List of tags to add.
+     */
+    public function addTags(array $tags)
+    {
+        $this->tags = array_merge($this->tags, $tags);
+    }
+
+    /**
+     * Set tags on the response.
+     *
+     * This must be called before any response is sent to the client.
+     * Make sure this is called after all addTags calls and only once.
+     */
+    public function flushTags()
```

Comment:
this is for the library only, in the bundle we will have something using Response.